src/decision/include/subroutines/bottom_line/FollowLine.h

```diff
@@ -0,0 +1,31 @@
+//
+// Created by drdc-s2632 on 17/07/18.
+//
+
+#ifndef PROJECT_FOLLOWLINE_H
+#define PROJECT_FOLLOWLINE_H
+
+#include "Subroutine.h"
+#include <line_detect/LineDetectMsg.h>
+
+/*** Communicating Class {findLine, adjustToLine, followLine} ***/
+class FollowLine : public Subroutine {
+  public:
+    FollowLine(std::unordered_map<std::string, double>* constants)
+      : Subroutine(constants) {}
```

Comment:
I'm not sure what these constants are, but passing in a pointer to a map seems incredibly error prone. (ie. this could compile fine but throw runtime errors). These should be explicit parameters (and stored as member variables) unless there is a special case I can't see here.

Reply:
This is an existing pattern from #50. DecisionNode pulls out a configuration and wants it to be given to all subroutines without copying the map to each one, this was my solution.

Reply:
These constants are from he .yaml file we import, was discussed in previous PR. I'm not sure what the proper was to do this was but I think this was the way we agreed on